src/_imaging.c

```diff
@@ -968,142 +977,172 @@ _convert_transparent(ImagingObject *self, PyObject *args) {
     return NULL;
 }
 
-static PyObject *
-_copy(ImagingObject *self, PyObject *args) {
-    if (!PyArg_ParseTuple(args, "")) {
-        return NULL;
-    }
-
-    return PyImagingNew(ImagingCopy(self->image));
+HPyDef_METH(Imaging_copy, "copy", Imaging_copy_impl, HPyFunc_NOARGS)
+static HPy Imaging_copy_impl(HPyContext *ctx, HPy self) {
+    PyObject *py_self = HPy_AsPyObject(ctx, self);
+    Imaging im = PyImaging_AsImaging(py_self);
+    return HPy_FromPyObject(ctx, PyImagingNew(ImagingCopy(im)));
 }
 
-static PyObject *
-_crop(ImagingObject *self, PyObject *args) {
+
+HPyDef_METH(Imaging_crop, "crop", Imaging_crop_impl, HPyFunc_VARARGS)
+static HPy Imaging_crop_impl(HPyContext *ctx, HPy self, HPy *args, HPy_ssize_t nargs) {
     int x0, y0, x1, y1;
-    if (!PyArg_ParseTuple(args, "(iiii)", &x0, &y0, &x1, &y1)) {
-        return NULL;
+    HPy h_tuple;
+    PyObject *tuple;
+
+    if (!HPyArg_Parse(ctx, NULL, args, nargs, "O", &h_tuple)) {
+        return HPy_NULL;
     }
 
-    return PyImagingNew(ImagingCrop(self->image, x0, y0, x1, y1));
+    tuple = HPy_AsPyObject(ctx, h_tuple);
+    
+    x0 = PyLong_AsLong(PyTuple_GetItem(tuple, 0));
```

Comment:
Is there a way to extract tuple items from an HPy tuple directly, without having to first convert it to a PyObject? I can't find any in the HPy source code